src/antsibull/collection_meta.py

```diff
@@ -13,50 +13,130 @@
 
 import os
 import typing as t
-from collections.abc import Mapping
 
+import pydantic as p
 from antsibull_core.yaml import load_yaml_file
+from packaging.version import Version as PypiVer
+from pydantic.functional_validators import BeforeValidator
+from typing_extensions import Annotated, Self
 
+if t.TYPE_CHECKING:
+    from _typeshed import StrPath
 
-class CollectionMetadata:
+
+def _convert_pypi_version(v: t.Any) -> t.Any:
+    if not isinstance(v, str):
+        raise ValueError(f"must be a string, got {v!r}")
+    if not v:
+        raise ValueError(f"must be a non-trivial string, got {v!r}")
+    version = PypiVer(v)
+    if len(version.release) != 3:
+        raise ValueError(
+            f"must be a version with three release numbers (e.g. 1.2.3, 2.3.4a1), got {v!r}"
+        )
+    return version
+
+
+PydanticPypiVersion = Annotated[PypiVer, BeforeValidator(_convert_pypi_version)]
+
+
+class RemovalInformation(p.BaseModel):
+    """
+    Stores metadata on when and why a collection will get removed.
+    """
+
+    model_config = p.ConfigDict(extra="ignore", arbitrary_types_allowed=True)
+
+    version: t.Union[int, t.Literal["TBD"]]
+    reason: t.Literal["deprecated", "considered-unmaintained", "renamed", "other"]
+    reason_text: t.Optional[str] = None
+    announce_version: t.Optional[PydanticPypiVersion] = None
+    new_name: t.Optional[str] = None
+    discussion: t.Optional[p.HttpUrl] = None
+    redirect_replacement_version: t.Optional[int] = None
```

Comment:
I think the key should also have a `_major_` in there. It's a different type than the other `_version` keys.

```suggestion
    redirect_replacement_major_version: t.Optional[int] = None
```

Reply:
Meh, it's only `announce_version` that's a full version number. I guess `version` could also be changed to `major_version`? What do you think?

Reply:
:+1: changed in 6b7271f2c174edf499d4b0e2fb57b022916140ef.